src/geff/utils.py

```diff
@@ -89,3 +89,38 @@ def validate(path: str | Path):
                         f"Edge property {prop} missing mask has length {missing_len}, "
                         f"which does not match id length {edge_id_len}"
                     )
+
+
+def check_equiv_geff(path_a, path_b):
+    """This function compares two geffs, typically a starting fixture geff with
+    the output of an implementation.
+
+    This tests focuses on maintaining shape and dtype consistency. It does not
+    assert element wise equality. path_a is assumed to be the "correct" geff.
+
+    Args:
+        path_a (str): Path to first zarr geff group
+        path_b (str): Path to second zarr geff group
+    """
+
+    za = zarr.open(path_a)
+    zb = zarr.open(path_b)
+
+    for graph_group in ["nodes", "edges"]:
+        ga = za[graph_group]
+        gb = zb[graph_group]
+
+        # Check ids
+        assert ga["ids"].shape == gb["ids"].shape
+        assert ga["ids"].dtype == gb["ids"].dtype
+
+        # Check that properties in each geff are the same
+        assert set(ga["props"]) == set(gb["props"])
+
+        # Check shape and dtype of each prop
+        for prop in ga["props"]:
+            if "missing" in ga[f"props/{prop}"]:
+                assert ga[f"props/{prop}/missing"].shape == gb[f"props/{prop}/missing"].shape
+                assert ga[f"props/{prop}/missing"].dtype == gb[f"props/{prop}/missing"].dtype
+            assert ga[f"props/{prop}/values"].shape == gb[f"props/{prop}/values"].shape
+            assert ga[f"props/{prop}/values"].dtype == gb[f"props/{prop}/values"].dtype
```

Comment:
Apologies if this is already addressed somewhere.  Won't this fail if there is no props on either of the graph_groups?

e.g `check_equiv_geff('mouse-20250719.zarr/tracks', 'mouse-20250719.zarr/tracks')` won't work. 

I've done:

```suggestion
        ga_has_props = "props" in ga
        gb_has_props = "props" in gb

        assert ga_has_props == gb_has_props

        if ga_has_props:
            # Check that properties in each geff are the same
            assert set(ga["props"]) == set(gb["props"])

            # Check shape and dtype of each prop
            # Check shape and dtype of each prop
            for prop in ga["props"]:
                if "missing" in ga[f"props/{prop}"]:
                    assert ga[f"props/{prop}/missing"].shape == gb[f"props/{prop}/missing"].shape
                    assert ga[f"props/{prop}/missing"].dtype == gb[f"props/{prop}/missing"].dtype
                assert ga[f"props/{prop}/values"].shape == gb[f"props/{prop}/values"].shape
                assert ga[f"props/{prop}/values"].dtype == gb[f"props/{prop}/values"].dtype
```

which I think is enough?

Reply:
Oh yes good catch! Thanks :)